dotCMS/src/integration-test/java/com/dotcms/storage/ContentletMetadataAPITest.java

```diff
@@ -0,0 +1,367 @@
+package com.dotcms.storage;
+
+import static com.dotcms.storage.StoragePersistenceProvider.DEFAULT_STORAGE_TYPE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.datagen.TestDataUtils.TestFile;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.collect.ImmutableSet;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import io.vavr.Tuple2;
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.FilenameUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+public class ContentletMetadataAPITest {
+
+    private static ContentletMetadataAPI contentletMetadataAPI;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        IntegrationTestInitService.getInstance().init();
+        contentletMetadataAPI = APILocator.getContentletMetadataAPI();
+    }
+
+    /**
+     * Given scenarios: We're testing metadata api against different types of asset-files
+     * Expected Results: we should get full and basic md for every type. Basic metadata must be included within the fm
+     * @throws IOException
+     */
+    @Test
+    @UseDataProvider("getFileAssetMetadataTestCases")
+    public void Test_Generate_Metadata_From_FileAssets(final TestCase testCase) throws IOException {
+
+        final String stringProperty = Config.getStringProperty(DEFAULT_STORAGE_TYPE);
+        try {
+            Config.setProperty(DEFAULT_STORAGE_TYPE, testCase.storageType.name());
+
+            //Remove any previously generated metadata by the checkin process
+            final File file = (File) testCase.fileAssetContent.get("fileAsset");
```

Comment:
Codacy found an issue: [The String literal "fileAsset" appears 5 times in this file; the first occurrence is on line 57](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)